salt/states/archive.py

```diff
@@ -1387,6 +1419,104 @@ def extracted(name,
             ret['comment'] = exc.strerror
             return ret
 
+    def _file_lists(src_path, dest_path):
+        '''
+        Return a tuple containing the file lists of given path for files, dirs, links
+        '''
+        ret = {
+            'files': set(),
+            'dirs': set(),
+            'links': set()
+        }
+        for root, dirs, files in salt.utils.path.os_walk(src_path, followlinks=True):
+            for fname in files:
+                src_file = os.path.join(root, fname)
+                dest_file = src_file.replace(src_path, dest_path.rstrip(os.sep))
+                if __salt__['file.is_link'](src_file):
+                    src_file_link = __salt__['file.readlink'](src_file, canonicalize=True)
+                    dest_file_link = src_file_link.replace(src_path, dest_path.rstrip(os.sep))
+                    cur_dest_file_link = __salt__['file.readlink'](dest_file, canonicalize=True)
+                    if not __salt__['file.is_link'](dest_file) or cur_dest_file_link != dest_file_link:
+                        ret['links'].add((dest_file, dest_file_link))
+                else:
+                    ret['files'].add((dest_file, src_file))
+            src_dir = root
+            dest_dir = root.replace(src_path, dest_path)
+            ret['dirs'].add((dest_dir, src_dir))
+        return ret['files'], ret['dirs'], ret['links']
+
+    if skip_same_file:
+        extracted_files = []
+        log.debug('Extracting %s to %s', name, origin_name)
+        mng_files, mng_dirs, mng_links = _file_lists(name, origin_name)
+        for dest_dir, src_dir in mng_dirs:
+            try:
+                if not os.path.isdir(dest_dir):
+                    if os.path.exists(dest_dir):
+                        __salt__['file.remove'](dest_dir)
+                    stats = __salt__['file.stats'](src_dir)
+                    __states__['file.directory'](
+                        dest_dir,
+                        user=stats.get('user'),
+                        group=stats.get('group'),
+                        file_mode=stats.get('mode')
+                    )
+            except Exception as exc:
+                __salt__['file.remove'](name)
+                ret['changes'] = {}
+                log.exception(traceback.format_exc())
```

Comment:
This is redundant. `log.exception()` will already append the traceback to the log message, and log at the `error` loglevel. The proper use of `log.exception()` would be to log an error message.